### PR TITLE
[Cmake] Make SYCL detection more portable

### PR DESCRIPTION
**Pull request by dmitriy-sobolev**
Let's use -fsycl only if it is supported by a compiler.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,26 +75,35 @@ include(CheckCXXCompilerFlag)
...
+            message(STATUS "SYCL is not supported")
+        endif()
+    endif()
 endif()
 
 # Set the default backend if one has not been explicitly provided
-if (NOT ONEDPL_BACKEND)
-    if (_sycl_support)
+if (NOT DEFINED ONEDPL_BACKEND)
+    if (SYCL_SUPPORT)
```
Comment (adamfidel) [inline CMakeLists.txt:106]: I'm confused where this variable `SYCL_SUPPORT` is coming from. Should this condition instead be checking if `FSYCL_OPTION_IF_SUPPORTED` is set to a non-empty value?

Comment (dmitriy-sobolev) [reply CMakeLists.txt:109]: Good catch. It was rather a refactoring mistake, which could be avoided by proper testing. Now `SYCL_SUPPORT` is set.

**Review by mmichel11 — APPROVED** [review]
This approach LGTM.

**Review by adamfidel — APPROVED** [review]
LGTM now as well!

**Pull request merged**